### ISSUE TITLE
Exception thrown for ipfs.key.list()

### ISSUE THREAD

**Issue by javaspeak** [ipfs-shipyard/java-ipfs-http-client#190]
Exception thrown for ipfs.key.list()

`io.ipfs.cid.Cid$CidEncodingException: Invalid cid bytes: 0172002408011220739f55f9afc90aa8f6529026941280b520652ff5c89b78e198b1bf37a5c8b280
	at io.ipfs.cid.Cid.cast(Cid.java:157)
	at io.ipfs.cid.Cid.decode(Cid.java:139)
	at io.ipfs.api.KeyInfo.fromJson(KeyInfo.java:42)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1624)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at io.ipfs.api.IPFS$Key.list(IPFS.java:236)`

When debugging it goes into this code:

`public static Codec lookup(long c) {
            if (!lookup.containsKey(c))
                throw new IllegalStateException("Unknown Codec type: " + c);
            return lookup.get(c);
        }`

The value of c is:  114

When debugging the json coming back, it looks fine and very similar to running the following on the command line:

json when debugging:

`{"Keys":[{"Name":"self","Id":"k51qzi5uqu5dj2bzd3tlqzpi3uwvikm261xax1recqzqu6zxx4x8gxyxvpalxc"},{"Name":"jd","Id":"k51qzi5uqu5dgc883u0idpp6u8iyv4rh7hxgubrpflgi6vhgq1miyab6v4ayn4"}]}`

`ipfs key list -l `

It is falling over on trying to parse the codec to get the CID.

Is this project supported anymore?

Should we use the code or is it dead and out of date?

Thank you

**Comment by javaspeak** [comment]
Noticed this ticket:

     https://github.com/ipfs-shipyard/java-ipfs-http-client/issues/182

Does that mean we need to wait for a new release for cid to work properly?

**Comment by javaspeak** [comment]
I cloned java-cid and changed pom to:

    v1.3.2

Then I ran:

    mvn package

Then I added the following to my pom.xml:

   <dependency>
         <groupId>com.github.ipld</groupId>
         <artifactId>java-cid</artifactId>
         <version>v1.3.2</version>
      </dependency>

I basically overid the version of cid dragged in via  java-ipfs-http-client and its  java-multiaddr

Now it works.

This has wasted some time.

Who can release a new release of java-cid with the master code and then update the poms of:

     java-multiaddr

and dependencies?

Basically the production releases are broken when master is not.  

Looks like someone forgot to do a release?
